packages/nimble-components/src/select/index.ts

```diff
@@ -635,14 +641,18 @@ export class Select
             }
         }
 
-        if (!this.open && this.selectedIndex !== currentActiveIndex) {
-            this.selectedIndex = currentActiveIndex;
-        }
+        if (!this.open || commitValueThenClose) {
+            if (this.selectedIndex !== currentActiveIndex) {
+                this.selectedIndex = currentActiveIndex;
+            }
 
-        if (!this.open && initialSelectedIndex !== this.selectedIndex) {
-            this.updateValue(true);
+            if (initialSelectedIndex !== this.selectedIndex) {
+                this.updateValue(true);
+            }
         }
 
+        this.open = commitValueThenClose ? false : this.open;
```

Comment:
nit: I'd vote for this since it removes ambiguity about whether setting `this.open` to its current value has any side effects. But admit it's at the cost of more lines so won't complain if you reject this suggestion.
```
if (comitValueThenClose) { 
    this.open = false; 
}
```

Reply:
+1 on being clearer about sideEffects. If no sideEffect is expected I like Jesse's suggestion. If one is expected we should add a comment about it